src/Features/Core/Portable/Diagnostics/DiagnosticService.cs

```diff
@@ -79,6 +79,37 @@ private void RaiseDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs args)
             }).CompletesAsyncOperation(eventToken);
         }
 
+        private void RaiseDiagnosticsCleared(object sender)
+        {
+            Contract.ThrowIfNull(sender);
+            var source = (IDiagnosticUpdateSource)sender;
```

Comment:
I'd move this to the caller and strongly type the parameter.

Reply:
sure.